tests/csapi/registration_test.go

```diff
@@ -0,0 +1,45 @@
+package csapi_tests
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"testing"
+
+	"github.com/matrix-org/complement/internal/b"
+	"github.com/matrix-org/complement/internal/client"
+	"github.com/matrix-org/complement/internal/match"
+	"github.com/matrix-org/complement/internal/must"
+	"github.com/tidwall/gjson"
+)
+
+func Test11Register(t *testing.T) {
```

Comment:
Please don't include 11 in this. Complement doesn't use the numbering scheme sytest does. We already have some registration tests so put this code as a subtest of the existing function please.